src/librustc/ty/mod.rs

```diff
@@ -491,6 +491,18 @@ pub struct TyS<'tcx> {
     region_depth: u32,
 }
 
+impl<'tcx> Ord for TyS<'tcx> {
+    fn cmp(&self, other: &TyS<'tcx>) -> Ordering {
+        self.sty.cmp(&other.sty)
+    }
+}
+
+impl<'tcx> PartialOrd for TyS<'tcx> {
+    fn partial_cmp(&self, other: &TyS<'tcx>) -> Option<Ordering> {
+        Some(self.sty.cmp(&other.sty))
```

Comment:
This could be `Some(self.cmp(other))`, to keep the implementation details in the `Ord` impl.